niworkflows/utils/misc.py

```diff
@@ -299,19 +299,24 @@ def check_valid_fs_license():
         FreeSurfer successfully executed (valid license)
 
     """
+    import os
     from pathlib import Path
     import subprocess as sp
     from tempfile import TemporaryDirectory
     from pkg_resources import resource_filename
 
+    env = os.environ.copy()
+    if lic is not None:
+        env["FS_LICENSE"] = os.path.abspath(lic)
```

Comment:
I would not bring this back. This makes the behavior of the function unpredictable.

Reply:
Can you elaborate? It seems very predictable to me. Unless you're concerned somebody's using a pre-6.0.1 version of FreeSurfer?

Reply:
IMHO, the canary should not alter the environment. Otherwise, it stops being usable in every context because you need to understand the environment.

Let's put the following example:

1. sMRIPrep runs in multiproc mode.
2. In the main process, but after the worker pool has been initialized, you check the license with the canary (with `lic` not `None`), `FS_LICENSE` was not set in the environment and the license is not under `FREESURFER_HOME`.
3. The canary succeeds because it accommodated its environment.
4. The first worker that needs a license fails, because the environment is different.

So, if the canary modifies the environment, you need to run the canary in every context it may be pertinent and always be careful of the order of operations.

Reply:
It is a problem of expectations: the canary should be a smoke test on the environment not a unit test of the license.

Reply:
I see. Then the purpose of it has changed. I would suggest that we deprecate the unit test capacity instead of removing it and making the test fail for downstream tools that still expect the result to be a meaningful assertion about a passed argument.

Would this be an acceptable compromise? It can be a very short deprecation window (i.e., remove completely in 1.3.0).